src/kernel/boot.c

```diff
@@ -580,13 +580,19 @@ BOOT_CODE static bool_t provide_untyped_cap(
     cap_t      root_cnode_cap,
     bool_t     device_memory,
     pptr_t     pptr,
-    word_t     size_bits,
-    seL4_SlotPos first_untyped_slot
```

Comment:
This commit only removes the first_untyped_slot field? I think keeping it explicitly helps people to understand the code. Overloading with untyped.end with two purposes is not as obvious as having a first_untyped_slot.

Reply:
The commit message could be worded more clearly.  The primary change is that `ndks_boot.bi_frame` is updated after each call to provide_untyped_cap with the updated region of untyped caps.  This change allows the code to check that no other caps are inserted between successive calls to provide_untyped_cap so that the final untyped region is correct.  It's not possible to check this without this change. Once this change has been made then first_untyped_slot becomes an unused parameter.

Reply:
The issue here is, that the whole concept only works out, if `ndks_boot.bi_frame->untyped.start` is properly setup. All untype caps must be in one "block". So my argument again that would be that passing a parameter `first_untyped_slot`explicitly creates an impression one can do something here - but that is not the case, if this gets out of sync this will cause strange effects eventually and we see an assert anyway.
Thus I think that not explicitly having this is better. However, what we could pass there is a pointer to the current boot info slot region that this is working on (`&ndks_boot.bi_frame->untyped`). Then `provide_untyped_cap()` and other would no really care which boot info slot region they are really working on, but could automatically update it. That is what I have tried out in https://github.com/axel-h/seL4/tree/patch-axel-67

Reply:
It would also be possible to keep the first_untyped_slot parameter and update `ndks_boot.bi_frame->untyped.end` inside the function.  Then you could still use asserts to catch if the ndks_boot.slot_pos_cur doesn't match the last call.